packages/wordcount/src/defaultSettings.js

```diff
@@ -1,3 +1,34 @@
+/** @typedef {import('./index').WPWordCountStrategy} WPWordCountStrategy */
+
+/** @typedef {Partial<{type: WPWordCountStrategy, shortcodes: string[]}>} WPWordCountL10n */
+
+/**
+ * @typedef WPWordCountSettingsFields
+ * @property {RegExp}              HTMLRegExp                        Regular expression that matches HTML tags
+ * @property {RegExp}              HTMLcommentRegExp                 Regular expression that matches HTML comments
+ * @property {RegExp}              spaceRegExp                       Regular expression that matches spaces in HTML
+ * @property {RegExp}              HTMLEntityRegExp                  Regular expression that matches HTML entities
+ * @property {RegExp}              connectorRegExp                   Regular expression that matches word connectors, like em-dash
+ * @property {RegExp}              removeRegExp                      Regular expression that matches various characters to be removed when counting
+ * @property {RegExp}              astralRegExp                      Regular expression that matches astral UTF-16 code points
+ * @property {RegExp}              wordsRegExp                       Regular expression that matches words
+ * @property {RegExp}              characters_excluding_spacesRegExp Regular expression that matches characters excluding spaces
+ * @property {RegExp}              characters_including_spacesRegExp Regular expression that matches characters including spaces
+ * @property {RegExp}              shortcodesRegExp                  Regular expression that matches WordPress shortcodes
+ * @property {string[]}            shortcodes                        List of all shortcodes
+ * @property {WPWordCountStrategy} type                              Describes what and how are we counting
+ * @property {WPWordCountL10n}     l10n                              Object with human translations
+ */
+
+/**
+ * Lower-level settings for word counting that can be overridden.
+ *
+ * @typedef {Partial<WPWordCountSettingsFields>} WPWordCountSettings
+ */
+
+/**
+ * @type {WPWordCountSettings}
+ */
```

Comment:
The inferred type may be better than the explicit type here. I'd consider omitting this:

```suggestion
```

With the annotation, all the properties are the `Partial<…>` type, so everything is possibly `undefined`. If we let the type be inferred, it can determine the type pretty well.

---

In TypeScript, I'd also consider a [`const` assertion](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions), but that seems to be unavailable via JSDoc (https://github.com/microsoft/TypeScript/issues/30445). We might be able to get something similar by using `Object.freeze` (for a runtime cost) or by playing with `@readonly` JSDoc tags on properties (not sure how well this works or applies).

Reply:
Thanks, @sirreal – done in ffb27cc. It did help with the narrowing the type.